rustls/tests/api.rs

```diff
@@ -6882,6 +6882,187 @@ impl<'a> io::Write for FakeStream<'a> {
     }
 }
 
+#[test]
+fn test_illegal_server_renegotiation_attempt_after_tls13_handshake() {
+    let client_config =
+        make_client_config_with_versions(KeyType::Rsa2048, &[&rustls::version::TLS13]);
+    let mut server_config = make_server_config(KeyType::Rsa2048);
+    server_config.enable_secret_extraction = true;
+
+    let (mut client, mut server) = make_pair_for_configs(client_config, server_config);
+    do_handshake(&mut client, &mut server);
+
+    let mut raw_server = RawTls::new_server(server);
+
+    let msg = PlainMessage {
+        typ: ContentType::Handshake,
+        version: ProtocolVersion::TLSv1_3,
+        payload: Payload::new(vec![
+            u8::from(HandshakeType::HelloRequest),
+            0x00,
+            0x00,
+            0x00,
+        ]),
+    };
+    raw_server.encrypt_and_send(&msg, &mut client);
+    let err = client
+        .process_new_packets()
+        .unwrap_err();
+    assert_eq!(format!("{err:?}"),
+               "InappropriateHandshakeMessage { expect_types: [NewSessionTicket, KeyUpdate], got_type: HelloRequest }");
```

Comment:
Will this be brittle over time? Would it be better to look for the presence of a key word or two instead of doing an exact match over the `Debug` repr?

Same comment applies to the other instances in this commit but I won't flag them each individually.

Reply:
replaced this and other instances  -- except the Message containing an alert, as `AlertMessagePayload` is not public and I'd like to avoid adding to `internal` exports.